src/domain/usecases/CheckSettingsPermissionsUseCase.ts

```diff
@@ -3,18 +3,19 @@ import { User } from "../../data/entities/User";
 import { UseCase } from "../../webapp/CompositionRoot";
 import { NamedRef } from "../entities/Ref";
 import { ConfigRepository } from "../repositories/ConfigRepository";
+import { Config } from "../entities/Config";
 
 export class CheckSettingsPermissionsUseCase implements UseCase {
     constructor(private configRepository: ConfigRepository) {}
 
-    public async execute(): Promise<boolean> {
+    public async execute(config?: Config): Promise<boolean> {
         const user = await this.configRepository.getUser();
-        const permissions = await this.configRepository.getSettingsPermissions();
+        const { settingsPermissions: permissions } = config ?? (await this.configRepository.get());
 
         const isAdmin = !!user.userRoles.find(role => role.authorities.find(authority => authority === "ALL"));
 
-        const sharedByUser = this.findCurrentUser(user, permissions.users ?? []);
-        const sharedByGroup = this.findCurrentUser(user, permissions.userGroups ?? []);
+        const sharedByUser = this.findCurrentUser(user, permissions?.users ?? []);
```

Comment:
We're checking many times through all the app with the null coalescing for `users` and `userGroups`. What are your thoughts on setting a default value directly to `[]` and removing the ?: from the `Permissions` type? Up to you 🙌🏼

Reply:
Yes, that's the same pattern we have seen, the less optional props we have in the code, the nicer working with these datastructures.